### PR TITLE
Fix PPA URL, add tools_only parameter, add start/stop checks to unit file

### PR DESCRIPTION
**Pull request by runejuhl**

### CODE REVIEW

File: manifests/packages.pp
Comment (flokli) [inline manifests/packages.pp]: I'm a bit unhappy with calling this `$version`. Maybe `$ensure` makes more sense, as it's already pretty common in some puppet resources.

**Review by flokli — CHANGES_REQUESTED** [review]

File: manifests/packages.pp
Comment (runejuhl) [reply manifests/packages.pp]: I'm fine with either. I'm not that fond of overloading `ensure` to set not just define whether a package should be present or not but also to control which version, but since it's common to use `ensure` -- let's keep it that way.

**Comment by runejuhl** [comment]
I've added some more minor changes to this PR -- if you want me to split it out into seperate PRs just let me know.

File: files/wireguard@.service
Comment (flokli) [inline files/wireguard@.service]: Hrm. I'm hoping to get rid of creating interfaces like this as soon as https://github.com/systemd/systemd/pull/4191 is ready.

In the meantime, I'd better like to use something like `ConditionPathExists=!/sys/class/net/%i`, if we want to ensure the interface doesn't yet exist. before starting the unit.

Can you rewrite all Execs to `/usr/bin/env $binary`?

File: manifests/packages.pp
Comment (flokli) [inline manifests/packages.pp]: typo in `wireguad`

**Review by flokli — CHANGES_REQUESTED** [review]

**Comment by flokli** [comment]
if you squash the requested changes into the existing commits, I'm fine with keeping things in the same PR :-)

File: files/wireguard@.service
Comment (runejuhl) [reply files/wireguard@.service]: Yeah, I'm waiting for that PR to be merged myself :)

Good idea about using `ConditionPathExists`, that didn't enter my mind. I'll update it in a sec.

**Comment by runejuhl** [comment]
Hi @flokli -- anything left before this can be merged?
(I've made the requested changes but since I squashed them into the existing commits the timestamps weren't updated so I'm not sure if you've gotten a message that the PR changed, even though the comments are now shown as "outdated"? ...I should probably just have reset timestamp :) )

**Comment by flokli** [comment]
@runejuhl yes, somehow I didn't receive an email about the changes. Merged now, thanks!

**Pull request merged**
